src/main/java/com/teragrep/aer_01/EventDataConsumer.java

```diff
@@ -0,0 +1,188 @@
+ * modifications.
+ *
+ * Supplemented terms under GNU Affero General Public License version 3
+ * section 7
+ *
+ * Origin of the software must be attributed to Suomen Kanuuna Oy. Any modified
+ * versions must be marked as "Modified version of" The Program.
+ *
+ * Names of the licensors and authors may not be used for publicity purposes.
+ *
+ * No rights are granted for use of trade names, trademarks, or service marks
+ * which are in The Program if any.
+ *
+ * Licensee must indemnify licensors and authors for any liability that these
+ * contractual assumptions impose on licensors and authors.
+ *
+ * To the extent this program is licensed as part of the Commercial versions of
+ * Teragrep, the applicable Commercial License may apply to this file if you as
+ * a licensee so wish it.
+ */
+package com.teragrep.aer_01;
+
+import com.azure.messaging.eventhubs.models.EventBatchContext;
+import com.codahale.metrics.Gauge;
+import com.codahale.metrics.MetricRegistry;
+import com.teragrep.aer_01.plugin.ParsedEventWithException;
+import com.teragrep.aer_01.plugin.WrappedPluginFactoryWithConfig;
+import com.teragrep.aer_01.records.ParsedEventListFromEventBatchFactory;
+import com.teragrep.aer_01.records.EventRecords;
+import com.teragrep.akv_01.event.ParsedEvent;
+import com.teragrep.akv_01.plugin.*;
+import com.teragrep.rlo_14.SDElement;
+import com.teragrep.rlo_14.SDParam;
+import com.teragrep.rlo_14.SyslogMessage;
+import com.teragrep.rlp_01.RelpBatch;
+import jakarta.json.JsonException;
+
+import java.nio.charset.StandardCharsets;
+import java.time.Instant;
+import java.util.List;
+import java.util.Map;
+import java.util.function.Consumer;
+import java.util.logging.Logger;
+import java.util.stream.Collectors;
+
+import static com.codahale.metrics.MetricRegistry.name;
+
+final class EventDataConsumer implements AutoCloseable, Consumer<EventBatchContext> {
+
+    private final Logger logger;
+    private final Output output;
+    private final Map<String, WrappedPluginFactoryWithConfig> pluginFactories;
+    private final MetricRegistry metricRegistry;
+    private final WrappedPluginFactoryWithConfig defaultPluginFactory;
+    private final WrappedPluginFactoryWithConfig exceptionPluginFactory;
+
+    EventDataConsumer(
+            final Logger logger,
+            final Output output,
+            final Map<String, WrappedPluginFactoryWithConfig> pluginFactories,
+            final WrappedPluginFactoryWithConfig defaultPluginFactory,
+            final WrappedPluginFactoryWithConfig exceptionPluginFactory,
+            final MetricRegistry metricRegistry
+    ) {
+        this.logger = logger;
+        this.metricRegistry = metricRegistry;
+        this.pluginFactories = pluginFactories;
+        this.exceptionPluginFactory = exceptionPluginFactory;
+        this.defaultPluginFactory = defaultPluginFactory;
+        this.output = output;
+    }
+
+    public void accept(final List<ParsedEvent> parsedEvents) {
```

Comment:
accept does not implement an interface however there are two methods with the same name which is confusing as the other implements one. the other and this seem not to share much as the other accesses no class members. should there be two objects instead?